packages/perseus/src/widgets/__tests__/passage_test.jsx

```diff
@@ -130,4 +167,48 @@ describe("passage widget", () => {
         // Assert
         expect(reference).toBe(null);
     });
+
+    it("should render passage title", () => {
+        renderPassage({passageTitle: "Passage title"});
+
+        expect(screen.getByText("Passage title")).toBeInTheDocument();
+    });
+
+    it("should render passage text", () => {
+        renderPassage({passageText: "Passage text"});
+
+        expect(screen.getByText("Passage text")).toBeInTheDocument();
+    });
+
+    it("should render footnotes", () => {
+        renderPassage({footnotes: "Footnote text"});
+
+        expect(screen.getByText("Footnote text")).toBeInTheDocument();
+    });
+
+    it("should render first question instructions", () => {
+        renderPassage({passageText: "[[test]] Passage text"});
+
+        expect(screen.getByText("The symbol")).toBeInTheDocument();
+        expect(screen.getAllByText("[Marker for question test]")).toHaveLength(
+            2,
+        );
+        expect(
+            screen.getByText(
+                "indicates that question test references this portion of the passage",
+            ),
+        ).toBeInTheDocument();
+    });
+
+    it("should render first sentence instructions", () => {
+        renderPassage({passageText: "[[1]] Passage text"});
+
+        expect(screen.getByText("The symbol")).toBeInTheDocument();
+        expect(screen.getAllByText("[Marker for question 1]")).toHaveLength(2);
```

Comment:
Question: Ooo, okay, I found this is described in passage-markdown, right? I understand there should be two of the squareLabels for each grouping of two square brackets, but I wasn't sure where in the code this is determined. Is it this line of passage-markdown (line 255): `match: SimpleMarkdown.inlineRegex(/^\[\[(\w+)\]\]( *)/),`?

Reply:
I honestly don't fully understand all of this, but the code you quoted is for finding custom Markdown syntax:
- `/^` start regex and match if the match is at the beginning of the string
- `\[\[` look for `[[` in the string
- `(w+)` capture any alphanumeric characters within the square brackets
- `\]\]` look for `]]` in the string
- `( *)` capture any spaces after the square brackets
- `/` end regex
 
So it will find `[[1]] ` and give us something like `['[[1]] ', '1', ' ']`. Then that match is passed to `parse` which does some logic and then passes the output of that logic to `react` for the rendering. Somewhere it's rendering it twice, I just don't understand how...although looking at this more, it looks like we might render this when we render the passage as a whole and then again in Passage's `_renderInstructions`.